File: Selenium/Project/Activity9.java
package OrangeHRM;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.Reporter;
import org.testng.annotations.*;
import java.time.Duration;
import java.util.List;

public class Activity9 {
        WebDriver driver;
        WebDriverWait wait;

        @BeforeTest(alwaysRun = true)
        public void setup() {
            // Set up Firefox driver
            WebDriverManager.firefoxdriver().setup();
            //Create a new instance of the Firefox driver
            driver = new FirefoxDriver();
            wait = new WebDriverWait(driver, Duration.ofSeconds(25));
            // Open the page
            driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");

            Reporter.log("Opened Browser |");
        }
        @Test(priority=0)
        public void loginTestCase() {
            //Find username and pasword fields
            WebElement usernameField = driver.findElement(By.id("txtUsername"));
            WebElement passwordField = driver.findElement(By.id("txtPassword"));

            //Enter values
            usernameField.sendKeys("orange");
            passwordField.sendKeys("orangepassword123");

            //Click Log in
            driver.findElement(By.id("btnLogin")).click();
            String pagetitle= driver.getTitle();
            System.out.println("pagetitle is: "+pagetitle);
            Assert.assertEquals(pagetitle, "OrangeHRM");
        }
        @Test(priority = 1)
        public void addemergencycontacts(){
            driver.findElement(By.id("menu_pim_viewMyDetails")).click();
            driver.findElement(By.xpath("//a[text()='Emergency Contacts']")).click();
            driver.findElement(By.id("btnAddContact")).click();
            driver.findElement(By.id("emgcontacts_name")).sendKeys("abc");
            driver.findElement(By.id("emgcontacts_relationship")).sendKeys("friend");
            driver.findElement(By.id("btnSaveEContact")).click();
            List<WebElement> alldetails=driver.findElements(By.xpath("//*[@id='frmEmpDelEmgContacts']"));
            System.out.println("All records: "+alldetails);
        }
        @AfterTest(alwaysRun = true)
        public void closeBrowser() {
            //Close the browser
            driver.close();
        }
    }

File: Selenium/Project/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
import time

# Set up the Firefox Driver with WebDriverManager
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the OrangeHRM page
    driver.get("http://alchemy.hguy.co/orangehrm")

    # Login with credentials
    username = driver.find_element(By.ID, "txtUsername")
    username.send_keys("orange")

    password = driver.find_element(By.ID, "txtPassword")
    password.send_keys("orangepassword123")

    login_btn = driver.find_element(By.ID, "btnLogin")
    login_btn.click()

driver.find_element(By.ID, "menu_pim_viewMyDetails").click()
driver.find_element(By.XPATH, "//a[text()='Emergency Contacts']").click()
driver.find_element(By.ID, "btnAddContact").click()
driver.find_element(By.ID, "emgcontacts_name").send_keys("abc")
driver.find_element(By.ID, "emgcontacts_relationship").send_keys("friend")
driver.find_element(By.ID, "btnSaveEContact").click()
alldetails = driver.find_elements(By.XPATH, "//*[@id='frmEmpDelEmgContacts']")
print("All records: ", alldetails)
driver.close()